src/NuGet.Core/NuGet.Configuration/Settings/SettingsFile.cs

```diff
@@ -108,15 +114,50 @@ public SettingsFile(string directoryPath, string fileName, bool isMachineWide, b
             IsMachineWide = isMachineWide;
             IsReadOnly = IsMachineWide || isReadOnly;
 
-            XDocument config = null;
-            ExecuteSynchronized(() =>
+            NuGetEventSource.Instance.Write(
+                ConfigurationSettingsFileRead,
+                new EventSourceOptions
+                {
+                    ActivityOptions = EventActivityOptions.Detachable,
+                    Keywords = NuGetEventSource.Keywords.Configuration,
+                    Opcode = EventOpcode.Start,
+                },
+                new
+                {
+                    ConfigFilePath,
+                    IsMachineWide = isMachineWide,
+                    IsReadOnly = isReadOnly
+                });
+
+            try
             {
-                config = FileSystemUtility.GetOrCreateDocument(CreateDefaultConfig(), ConfigFilePath);
-            });
+                XDocument config = null;
+                ExecuteSynchronized(() =>
+                {
+                    config = FileSystemUtility.GetOrCreateDocument(CreateDefaultConfig(), ConfigFilePath);
+                });
 
-            _xDocument = config;
+                _xDocument = config;
 
-            _rootElement = new NuGetConfiguration(_xDocument.Root, origin: this);
+                _rootElement = new NuGetConfiguration(_xDocument.Root, origin: this);
+            }
+            finally
+            {
+                NuGetEventSource.Instance.Write(
+                    ConfigurationSettingsFileRead,
+                    new EventSourceOptions
+                    {
+                        ActivityOptions = EventActivityOptions.Detachable,
+                        Keywords = NuGetEventSource.Keywords.Configuration,
+                        Opcode = EventOpcode.Stop,
+                    },
+                    new
+                    {
+                        ConfigFilePath,
```

Comment:
The anonymous type used for the stop event's data appears to be the same as the start event's data. If we keep this as a class, that gets a heap allocation, we should at least re-use the same instance.

Reply:
Thanks, I've changed the pattern to use a private class for event methods and a `record struct` for event data.  I think the code looks much better now.